src/OrchardCore.Modules/OrchardCore.Settings/Recipes/SettingsStep.cs

```diff
@@ -23,80 +24,115 @@ protected override async Task HandleAsync(RecipeExecutionContext context)
         var model = context.Step;
         var site = await _siteService.LoadSiteSettingsAsync();
 
+        if (model == null)
+        {            
+            return;
+        }
+
+        site ??= new SiteSettings();
+
         foreach (var property in model)
         {
+            var value = property.Value;
```

Comment:
Here add


if (property.Value is null)
{
    continue;
}


Then remove all the null check thereafter